pkg/reconciler/managed/reconciler.go

```diff
@@ -138,6 +144,11 @@ func (fn CriticalAnnotationUpdateFn) UpdateCriticalAnnotations(ctx context.Conte
 // resource, for example usernames, passwords, endpoints, ports, etc.
 type ConnectionDetails map[string][]byte
 
+// AdditionalDetails represent any additional details the external client wants
+// to return about an operation that has been performed. These details will be
+// included in the change logs.
+type AdditionalDetails map[string]any
```

Comment:
Should the values of this map really be as open-ended as `any`? Could we get away with requiring string values?

Reply:
I was thinking the flexibility would be nice for externalClients to be able to return JSON-y type of details (e.g. `map[string]object`), but I'm not a full 100% on that.

What's the biggest benefit you think we'd get from imposing a more constrained type of `map[string]string`?

Reply:
It's simpler to reason about, and eliminates any issues with folks trying to return values that can't easily be serialized. Unless there's a clear reason to support arbitrary/complex values I'd keep it simple.

Reply:
Cool! that's reasonable and I can make that change. The more PRs I review, the more I appreciate things being simple to reason about 😅

Reply:
Done! Additional details are now constrained to `map[string]string` in https://github.com/crossplane/crossplane-runtime/pull/754/commits/afb1c6b1148b69d12412523e3133266c8c810ea2.

Note I also realized that `Delete()` didn't have the ability to specify additional details because it currently has no return type - it can only return errors. I also updated that with https://github.com/crossplane/crossplane-runtime/pull/754/commits/afb1c6b1148b69d12412523e3133266c8c810ea2#diff-97300a2543f95f5a2ada3560bf47dd7334e237e27976574d15d1cddef2e66c01R345, but let me know if you'd like to skip that because it does force consumers of the managed reconciler to update to this new `Delete()` signature whenever they update their crossplane-runtime dependency next, a la https://github.com/jbw976/provider-kubernetes/commit/4a9146acc2b2444517ccc66f43cb3a2aa650a2e1#diff-475eea76845069ccefe80a5e3e03cf95a235099281bb9bcc874d19f243ed8e2c.